dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/JarLinkageReport.java

```diff
@@ -148,4 +148,25 @@ public JarLinkageReport reachableErrors() {
     builder.setJarPath(getJarPath());
     return builder.build();
   }
+
+  public String getErrorString() {
+    String indent = "  ";
+    StringBuilder builder = new StringBuilder();
+    int totalErrors = getCauseToSourceClassesSize();
+
+    builder.append(getJarPath().getFileName() + " (" + totalErrors + " errors):\n");
+    for (SymbolNotResolvable<ClassSymbolReference> missingClass : getMissingClassErrors()) {
```

Comment:
Only if you want, this can be condensed:

```java
String.format("%s (%s errors):\n%s\n",
    getJarPath().getFileName(),
    totalErrors,
    Stream.of(getMissingClassErrors(), getMissingMethodErrors(), getMissingFieldErrors())
        .flatMap(Collection::stream)
        .map(error -> indent + error.getReference().getErrorString() + "\n")
        .collect(joining("\n"));
```

Or you could use that stream with a `forEachOrdered(builder::append)` if you want.

Or you could use Guava's `Iterables.concat(getMissingClassErrors(), getMissingMethodErrors(), getMissingFieldErrors())` and a single for loop.